### DISCUSSION TITLE
Package Pruning

### DISCUSSION THREAD

**Discussion by AnthonyNeo** [atom/atom#23604] [Ideas]
### The Package Library _needs_ a freshmaker

For every possible tool search, there's at least a dozen package results, and most of them are defunct. This alone was enough to bring me here to find a package to clean out the package results. Even that is a sisyphean task of sifting through hundreds of irrelevant or defunct packages, checking the repo for activity, looking for replacements/alts.

What I find, is a community full of people running into brick walls of abandoned and broken packages. Packages that no longer function, haven't been touched in over 5 years, but are still being included in the package list. 

As a result, I feel forced to VSCode, though I _really_ want to stay with opensource, and even help improve it. 

**Bear Minimum**
Add "last updated" to package listings, and allow filtering by that date.

**Better**
Filter results by explicit Atom version in `package.json`.
* `"atom": ">0.4"` --> 0.4
* `"atom": ">1.0 <2.0"` --> 1.0
* `"atom": "<2.0"` --> 0.0
This will even help enforce better packaging. But, What to do with it?
* Matches current version --> "RECENT"
* <= 3 subversion ago --> (included in results)
* 3 to 10 subversions ago --> "OUTDATED"
* > 10 subversions or 24 months --> (excluded from results)

**Best**
Some basic test requirements. Must include 1 unit or integration test to be run in an Atom automated test. Last Atom version with a passed test is recorded in package.json. Only show packages with a passed test from the current Atom version.

**Comment by keevan** [comment]
Also once a name is taken it's taken forever :'( which means packages that are outdated are replaced with packages with similar names, and so on. Then you have a bunch of packages that might work or might not work. But such is the nature of open source in a way. It's also difficult to know if something would still work regardless of how old something is. So to an extent I feel your pain.

**Comment by nlydv** [comment]
Agreed. I'm pretty determined not to switch to another editor for many reasons, but mostly because of how open and hackable Atom is... which recently (and sorta unintentionally) led me to creating two replacements for broken/outdated packages myself:

- [`auto-sass`](/nlydv/auto-sass)
- [`pigments-redux`](/nlydv/pigments-redux) (_small fork of [pigments](/abe33/atom-pigments) just to clear away obvious bugs_)

It would be great if there was a way to properly sift through all the highly popular yet broken packages to find newer & better working ones. I feel like there were probably some out there that I missed but would've worked just fine for me, perhaps I went about DIY-ing my own packages for no reason `¯\_(ツ)_/¯`
